### DISCUSSION TITLE
Architecture question - load balancer installed on all WEB API instances, but enabled only for one

### DISCUSSION THREAD

**Discussion by SorinMiron** [dotnet/yarp#2274] [Q&A] [answered]
Hello folks,

I searched a lot trough threads but I cannot find the question for the following scenario:

Suppose  there are 2 WEB API instances: WebApi1 and WebApi2 which are the same. The Yarp.Reverse proxy is installed in both instances - but only enabled in one. 
Specific case:
WebApi1 has an endpoint (/location) and a load balancer that will balance the load between WebApi1 and WebApi2
WebApi2 has the same endpoint (/location) and a load balancer that is not configured at all.

Is it possible to load balance between WebApi1 and WebApi2 when the gateway is embedded in WebApi1?

**Comment by Tratcher** [comment, marked as answer]
In theory, yes, but not with the built in load balancer, only with the IHttpForwarder.

Say you want WebApi1 to handle 50% of traffic at endpoint "/location" and forward the other 50%. At the start of that endpoint you add code to check if load balancing is enabled for this instance. If it is, decide if this request should be forwarded. If so, call IHttpForwarder, otherwise let the endpoint run as usual.

This only makes sense if the endpoint logic is significantly more expensive than the forwarding logic. Otherwise WebApi1 is doing twice as much work just to forward half of the requests. For truly balanced load you'd need a 3rd instance that only forwards to 1 or 2.

**Reply by SorinMiron** [reply]
Thanks. I'll go with the 3rd instance instead. Seems to be the best practice
